lib/most_frequent_integer.rb

```diff
@@ -3,5 +3,52 @@
 # exists, return nil.
 
 def find_most_frequent_integer(arr)
-  # put your code here
+
+sortarr = arr.sort
+indexarr = Array.new
+n = 0
+
+if sortarr.empty?
+    return nil
+else
+    i = sortarr[0]
+end
+
+while n < sortarr.length && i <= sortarr.last
+
+    indexarr[n] = sortarr.each_index.select{|x| sortarr[x] == i} 
+    i = i + 1
+    n = n + 1
+    
 end
+
+m = 0
+p = 0
+q = p + 1
+
+while m < indexarr.size && q < indexarr.size
+    if p == q
```

Comment:
I only put it in there because I wanted to prevent a trial with a code that would produce p = q mistakenly. My intention is for p to represent the largest sub array in indexarr and the thought of starting p at the 0th spot and if q were to proceed down indexarr only to find a larger sub array, then I would switch the value of p to q follwed by resetting q to 0 causing the loop to start over.

As I wrote that I now realize how silly it is to reset q to the very first spot when it should have continued after changing p to the index where the new largest sub array exists. But, because the code has q starting over again, q can indeed encounter p again since its starting at zero in the above scenario.